core/controllers/learner_dashboard.py

```diff
@@ -303,3 +304,52 @@ def get(self) -> None:
                 learner_dashboard_activities.to_dict())
         })
         self.render_json(self.values)
+
+
+class LearnerExplorationProgressHandlerNormalizedRequestDict(TypedDict):
+    """Dict representation of LearnerExplorationProgressHandler's
+    normalized_request dictionary.
+    """
+
+    exp_ids: List[str]
+
+
+class LearnerExplorationProgressHandler(
+    base.BaseHandler[
+        Dict[str, str],
+        LearnerExplorationProgressHandlerNormalizedRequestDict
+    ]
+):
+    """Handles fetching progress of a user in all exploration"""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS: Dict[str, str] = {}
+
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {
+            'exp_ids': {
+                'schema': {
+                    'type': 'custom',
+                    'obj_type': 'JsonEncodedInString'
+                }
+            }
+        }
+    }
+
+    # TODO(#16539): Change the type of value that is rendered to JSON,
```

Comment:
Hi, @seanlip Thanks for the review,

- I need to take all three variables (exp_id, visited checkpoints, total checkpoints). Because I am using them in the front end. Also, It is already used in many places in the code base please see here (https://github.com/SubhashThenua/oppia/blob/develop/core/controllers/learner_group.py#L1072).
Thanks!

Reply:
Yes, read my comment more carefully please. It still allows you to send all the information you want. I'm talking about adding an outer dict to the response, i.e.

```
{
    'exploration_progress': exploration_progress
}
```

so that you don't need to introduce a TODO.

Also, just because a TODO exists elsewhere is not a reason to add more TODOs in new code that you write. That just ends up leaving more stuff for people to clean up. So please don't do it in future.